internal/handlers/pg/msg_listcollections.go

```diff
@@ -93,20 +90,37 @@ func (h *Handler) MsgListCollections(ctx context.Context, msg *wire.OpMsg) (*wir
 			continue
 		}
 
+		if nameOnly {
+			d = must.NotFail(types.NewDocument(
+				"name", n,
+			))
+		}
+
 		collections.Append(d)
 	}
 
 	var reply wire.OpMsg
-	must.NoError(reply.SetSections(wire.OpMsgSection{
-		Documents: []*types.Document{must.NotFail(types.NewDocument(
-			"cursor", must.NotFail(types.NewDocument(
-				"id", int64(0),
-				"ns", db+".$cmd.listCollections",
+
+	switch {
+	case nameOnly:
+		must.NoError(reply.SetSections(wire.OpMsgSection{
+			Documents: []*types.Document{must.NotFail(types.NewDocument(
 				"firstBatch", collections,
```

Comment:
I think tests fail because we still need to return a fake cursor there

Reply:
I updated it and it works. Thanks!